__tests__/integration/utils/wallet-helper.js

```diff
@@ -39,30 +53,87 @@ export class WalletHelper {
     return this.#addresses;
   }
 
+  get started() {
+    return this.#started;
+  }
+
   /**
-   * Creates a wallet object but does not start it on server
-   * @param {string} walletId
-   * @param {string} [words] 24 words
+   * Starts all the wallets needed for the test suite.
+   * <b>This is the preferred way of starting wallets</b> on the Integration Tests,
+   * performance-wise.
+   * @param {WalletHelper[]} walletsArr Array of WalletHelpers
+   * @param [options]
+   * @param {boolean} [options.skipAddresses] Skips the getSomeAddresses command
+   * @param {number} [options.amountOfAddresses=10] How many addresses should be cached per wallet
+   * @returns {Promise<void>}
    */
-  constructor(walletId, words) {
-    if (!walletId) throw new Error(`Wallet must have a walletId`);
-    this.#walletId = walletId;
+  static async startMultipleWalletsForTest(walletsArr, options) {
+    /**
+     * A map of `WalletHelper`s indexed by their `walletId`s
+     * @type {Record<string,WalletHelper>}
+     */
+    const walletsPendingReady = {};
 
-    this.#words = words || TestUtils.generateWords();
+    // If the genesis wallet is not instantiated, start it. It should be always available
+    const { genesis } = WALLET_CONSTANTS;
+    const isGenesisStarted = await TestUtils.isWalletReady(genesis.walletId);
+    if (!isGenesisStarted) {
+      walletsArr.unshift(new WalletHelper(genesis.walletId, genesis.words));
+    }
+
+    // Requests the start of all the wallets in quick succession
+    for (const wallet of walletsArr) {
+      await TestUtils.startWallet({
+        walletId: wallet.walletId,
+        words: wallet.words,
+      });
+      walletsPendingReady[wallet.walletId] = wallet;
+    }
+
+    // Enters the loop checking each wallet for its status
+    while (true) {
+      const pendingWalletIds = Object.keys(walletsPendingReady);
+      if (!pendingWalletIds.length) {
```

Comment:
Maybe we should also add a timeout on this loop

I guess that if the wallet headless loses connection to the full-node, it will return not ready indefinitely

Reply:
Implemented a timeout option with a default to 60 seconds, adjustable via config file.
Also added some logging to help identify if any test suite makes this fail. It showed me that currently the wallet initializations are between 8 - 47 seconds.